### ISSUE TITLE
Setting a Target exe does not appear to work. 

### ISSUE THREAD

**Issue by zmarotrix** [hannesmann/vanillafixes#67]
Neither dragging a modified exe nor creating a shortcut seem to work. Both give the following error. 

![image](https://github.com/user-attachments/assets/226b7938-b610-43c4-b73e-c433135c8661)

Shortcut has ```"G:\Games\WoW Private Server\Turtle WoW\VanillaFixes.exe" WoW_tweaked.exe```and is in the same folder.

**Comment by Azzc0** [comment]
Your working directory is important. You'll want to add the full path on the target binary.
Try this instead:
`"G:\Games\WoW Private Server\Turtle WoW\VanillaFixes.exe" "G:\Games\WoW Private Server\Turtle WoW\WoW_tweaked.exe"`

**Comment by zmarotrix** [comment]
Same result. The shortcut is in the same folder as the game and the VanillaFixes.exe/
![image](https://github.com/user-attachments/assets/748463a0-9ee6-4743-9a5a-870b457aa838)

**Comment by shikulja** [comment]
same issue https://github.com/hannesmann/vanillafixes/issues/60

original WoW.exe need in game folder

**Comment by zmarotrix** [comment]
Out of curiosity, I made a fake WoW.exe file (renamed empty text document) and it seems to launch when I use it. So it seems to be checking to see if WoW.exe exists before checking the argument.

**Comment by zmarotrix** [comment]
Issue appears to be in line 42 of `main.c`.

This check needs to be updated to only happen if the shortcut does not include a new EXE. This check seems to happen a bit lower on line 94. I'm not sure I understand this well enough to modify it, otherwise I'd submit a pr.

**Comment by zmarotrix** [comment]
Maybe I'm misunderstanding or making some assumptions here, but I believe this should solve the issue. 
```
if (cmdLineData.pWowExePath == NULL) {
    cmdLineData.pWowExePath = malloc(MAX_PATH * sizeof(WCHAR));
    PathCombine(cmdLineData.pWowExePath, pWowDirectory, L"WoW.exe");
}

// Use the parsed executable path
LPWSTR pWowExePath = cmdLineData.pWowExePath;
```

**Comment by pojunen** [comment]
Yeah this needs some changes for sure, I've tried on Windows too (my main OS is linux) but couldn't get WoW_tweaked.exe to work along with vanillafixes, I've tried everything possible but either:

I get OP error, or I can't get pass on login screen (Unable to validate game version error)

**Comment by hannesmann** [comment]
The reason why VanillaFixes always checks for WoW.exe in the game folder is because there are some servers that kick you out if WoW.exe is missing (or has the wrong checksum). You can still use vanilla-tweaks and VanillaFixes on those servers, but WoW.exe needs to stay unchanged.

> Or new fork which would include features from both vanillafixes and vanilla-tweaks in one package. I guess this would be optimal option for now as both vanillafixes and vanilla-tweaks are quite inactive

I also agree that a fork is the best option :) I don't play WoW anymore and I'm not really planning on adding any more features to VanillaFixes. If someone wants to create a user friendly all-in-one package I would say go for it.
